jdsol-spec/src/main/asciidoc/jdsol.adoc

```diff
@@ -1,6 +1,1240 @@
+|https://docs.oracle.com/javase/1.5.0/docs/guide/jpda/jdi/com/sun/jdi/Method.html#locationsOfLine(int)[`List locationsOfLine(int lineNumber)`]
+
+2+^|https://docs.oracle.com/javase/1.5.0/docs/guide/jpda/jdi/index.html[JDI] -
+https://docs.oracle.com/javase/1.5.0/docs/guide/jpda/jdi/com/sun/jdi/Location.html[`Location`] interface
+
+|
+|https://docs.oracle.com/javase/1.5.0/docs/guide/jpda/jdi/com/sun/jdi/Location.html[class comment] (strata defined)
+
+|https://docs.oracle.com/javase/1.5.0/docs/guide/jpda/jdi/com/sun/jdi/Location.html#lineNumber(java.lang.String)[`int lineNumber(String stratum)`]
+|https://docs.oracle.com/javase/1.5.0/docs/guide/jpda/jdi/com/sun/jdi/Location.html#lineNumber()[`int lineNumber()`]
+
+|https://docs.oracle.com/javase/1.5.0/docs/guide/jpda/jdi/com/sun/jdi/Location.html#sourceName(java.lang.String)[`String sourceName(String stratum)`]
+|https://docs.oracle.com/javase/1.5.0/docs/guide/jpda/jdi/com/sun/jdi/Location.html#sourceName()[`String sourceName()`]
+
+|https://docs.oracle.com/javase/1.5.0/docs/guide/jpda/jdi/com/sun/jdi/Location.html#sourcePath(java.lang.String)[`String sourcePath(String stratum)`]
+|
+
+|https://docs.oracle.com/javase/1.5.0/docs/guide/jpda/jdi/com/sun/jdi/Location.html#sourcePath()[`String sourcePath()`]
+|
+
+|===
+
+== SourceDebugExtension Support
+
+Debugger applications frequently need debugging information about the source
+that exceeds what is delivered by the existing Java^TM^ Virtual Machine class
+file attributes (SourceFile, LineNumber, and LocalVariable). This is
+particularly true for debugging the source of other languages. In a distributed
+environment side files may not be accessible, the information must be directly
+associated with the class.
+
+The solution is the addition of a class file attribute which holds a string.
+The string contains debugging information in a standardized format which allows
+for evolution and vendor extension.
+
+=== SourceDebugExtension Access
+
+This string is made opaquely accessible at the three layers of the
+https://docs.oracle.com/javase/1.5.0/docs/guide/jpda/[Java Platform Debugger Architecture] (JPDA):
+
+[cols="10%,90%"]
+|===
+|JVMDI |https://docs.oracle.com/javase/1.5.0/docs/guide/jpda/jvmdi-spec.html#GetSourceDebugExtension[`GetSourceDebugExtension(jclass clazz, char **sourceDebugExtensionPtr)`]
+|JDWP |https://docs.oracle.com/javase/1.5.0/docs/guide/jpda/jdwp/jdwp-protocol.html#JDWP_ReferenceType_SourceDebugExtension[`SourceDebugExtension`] Command (12) in the `ReferenceType` (2) Command Set
+|JDI |https://docs.oracle.com/javase/1.5.0/docs/guide/jpda/jdi/com/sun/jdi/ReferenceType.html#sourceDebugExtension()[`String sourceDebugExtension()`] in the `ReferenceType` interface
+|===
+
+[#attribute]
+=== `SourceDebugExtension` Class File Attribute
```

Comment:
This is a heading, drop the source/code/pre formatting on `SourceDebugExtension` (you don't do this elsewhere on headings)

Reply:
Done. It was only there because it came across in the conversion from the original.